src/SoftwareSerial.h

```diff
@@ -32,11 +32,76 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+	SWSERIAL_6O1 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_7O1 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_8O1 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_5M1 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_6M1 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_7M1 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_8M1 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_5S1 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_6S1 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_7S1 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_8S1 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_1 ),
+	SWSERIAL_5N2 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_NONE  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_6N2 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_NONE  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_7N2 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_NONE  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_8N2 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_NONE  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_5E2 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_EVEN  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_6E2 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_EVEN  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_7E2 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_EVEN  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_8E2 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_EVEN  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_5O2 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_6O2 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_7O2 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_8O2 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_ODD   | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_5M2 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_6M2 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_7M2 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_8M2 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_MARK  | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_5S2 = ( SWSERIAL_NB_BIT_5 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_6S2 = ( SWSERIAL_NB_BIT_6 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_7S2 = ( SWSERIAL_NB_BIT_7 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_2 ),
+	SWSERIAL_8S2 = ( SWSERIAL_NB_BIT_8 | SWSERIAL_PARITY_CONF_SPACE | SWSERIAL_NB_STOP_BIT_2 )
+};
+
+enum ParityMode : uint8_t
```

Comment:
I wouldn't mind a different name for Parity - mode is one use, actual value another, I find mode confusing. Parity, ParityValue, SerialParity. By the way, how about moving all the supporting types into class scope, in lieu of using namespaces proper this solves the problem of poluting the global namespace with conflicts.